### PR TITLE
Scroll element into view first + lint files + getRandomPagePoint

### PR DESCRIPTION
**Pull request by Niek**
When the element is out of view, clicking doesn't work. This fixes this issue by scrolling the element into view first.

Test case:
```javascript
    const cursor = createCursor(page);
    await page.goto('https://old.reddit.com/');
    const element = '.footer a';
    await page.waitForSelector(element);
    await cursor.click(element);
```

### CODE REVIEW

**Comment by Niek** [comment]
I see my eslint changed the format quite a bit. Do you want me to change it back (or do you have your own preferred `.eslintrc`) @Xetera?

File: src/spoof.ts [hunk 1]
```diff
@@ -87,45 +75,42 @@ export const createCursor = (page: Page, start: Vector = origin) => {
...
         throw new Error(
-          `Could not find element with selector "${selector}", make sure you're waiting for the elements with "puppeteer.waitForSelector"`
+          `Could not find element with selector "${selector}", make sure you're waiting for the elements with "puppeteer.waitForSelector"`,
         );
       }
+      // Make sure the object is in view
+      let objectId;
+      if ((objectId = (elem as any)._remoteObject.objectId)) {
+        await (page as any)._client.send('DOM.scrollIntoViewIfNeeded', {
+          objectId,
```
Comment (Xetera) [inline src/spoof.ts:85]: Looks like [scrollintoViewIfNeeded](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoViewIfNeeded) is deprecated. Can you also change this up so it doesn't use private properties on puppeteer?

Comment (Niek) [reply src/spoof.ts:85]: You're looking at the wrong documentation, that's the DOM scrollIntoViewIfNeeded call. This PR is using the CDP call instead: https://chromedevtools.github.io/devtools-protocol/tot/DOM/#method-scrollIntoViewIfNeeded. That's the only way to do this without detection.

Comment (Niek) [reply src/spoof.ts:85]: The only other way to call CDP methods directly is by creating a new CDP session: https://pptr.dev/#?product=Puppeteer&version=v5.4.1&show=api-class-cdpsession

That's more "type safe" but also more overhead as it creates an extra session.

Comment (Niek) [reply src/spoof.ts]: I added eslint config, `yarn lint` support and a `getRandomPagePoint()` function as well so it doesn't always start at 0,0

Comment (Niek) [reply src/spoof.ts]: Also switched to puppeteer-core for smaller dependencies and updated the types

**Pull request merged**